File: 11/main.py
def transform(stone):

    if stone == 0:
        return [1]
    string_stone = str(stone)
    if len(string_stone) % 2 == 0:
        mid = len(string_stone)//2
        return [int(string_stone[:mid]), int(string_stone[mid:])]
    return [stone * 2024]


fname = "input.txt"

with open(fname) as f:
    input_string = f.read().strip()
    stones = map(int, input_string.split(" "))

total = 0
for j, stone in enumerate(stones):
    temp = [stone]
    for i in range(25):
        temp2 = []
        for s in temp:
            ans = transform(s)
            temp2 += ans
        temp = temp2
    total += len(temp)

print(total)

File: 11/main.js

function transform(stone){
    if (stone === 0){
        return [1]
    }
    const string_stone = stone.toString()
    if (string_stone.length % 2 === 0){
        const mid = Math.floor(string_stone.length / 2)
        return [Number(string_stone.slice(0,mid)), Number(string_stone.slice(mid))]
    }
    return [stone * 2024]
}

const path = "input.txt";
const file = Bun.file(path);

const text = await file.text();
let stones = text.split(' ').map((s)=>Number(s))

for (let i=0; i<75;i++){
    console.log(i)
    let temp = []
    for (const stone of stones){
        let ans = transform(stone)
        temp = temp.concat(ans)
    }
    stones = temp
}

console.log(stones.length)
